01_Data/src/layers/sequelize/repository/DeviceModel.ts

```diff
@@ -346,10 +346,10 @@ export class SequelizeDeviceModelRepository extends SequelizeRepository<Variable
 
   async findComponentAndVariable(componentType: ComponentType, variableType: VariableType): Promise<[Component | undefined, Variable | undefined]> {
     const component = await this.component.readOnlyOneByQuery({
```

Comment:
in addition the `undefined` issue fixed here, we noticed `readOnlyOneByQuery` would fail if it found two component entries, which happened because i had an EVSE with two connectors, and therefore each component entry had a different `evseDatabaseId` 🤔 

i see down below in the transaction module that problem is solved in there -- would a similar logic need to be applied to additionally search by evseId/connectorId (if that's even possible here) to prevent this issue or is that a concern for a separate PR?

Reply:
> which happened because i had an EVSE with two connectors, and therefore each component entry had a different evseDatabaseId

This is a protocol breach, so we don't need to worry about that case. Two components with the same name and instance cannot exist.

This method does not support searching for a component by name only to find different instances. Undefined is treated as null, so there is no argument that can be given which would do that. This is why I had to resort to a query directly on an instance of CrudRepository<Component> to get what we need in the StatusNotification handler.